src/request-api.js

```diff
@@ -90,9 +90,11 @@ function requestAPI (config, path, args, qs, files, buffer, cb) {
   // this option is only used internally, not passed to daemon
   delete qs.followSymlinks
 
+  const port = config.port ? `:${config.port}` : ''
+
   const opts = {
     method: files ? 'POST' : 'GET',
-    uri: `${config.protocol}://${config.host}:${config.port}${config['api-path']}${path}?${Qs.stringify(qs, {arrayFormat: 'repeat'})}`,
+    uri: `${config.protocol}://${config.host}${port}${config['api-path']}${path}?${Qs.stringify(qs, {arrayFormat: 'repeat'})}`,
```

Comment:
Handling this with the `url` module is probably best. It cleans up that template a lot. Not a huge issue though.

```javascript
let uri = url.format({
  protocol: config.protocol,
  hostname: config.host,
  port: config.port,
  pathname: `${config['api-path']}/${path}`.replace(/\/+/g, '/'), // Double slashes don't matter, but it looks nicer.
  query: qs
})
```

It will handle the undefined port and not include it.